src/pgrn-custom-scan.c

```diff
@@ -1,6 +1,58 @@
+#include "pgrn-custom-scan.h"
 #include "pgroonga.h"
 
+typedef struct PGrnScanState
+{
+	CustomScanState customScanState; /* must be first field */
```

Comment:
How about renaming this to `parent`?

```suggestion
	CustomScanState parent; /* must be first field */
```

Reply:
I have defined `PGrnScanState` to add a PGroonga specific field to `CustomScanState`.
(There are no specific fields yet, but we will add them in the implementation.)
So I think `customScanState` (or `state` or something) would be better, since it should be the current state, not the parent, what do you think?

Reply:
In general, we can use cast to access parent struct in C:

```c
PGrnScanState *state = palloc(sizeof(PGrnScanState));
CustomScanState *base_state = (CustomScanState *)state;
base_state->XXX = ...;
```

We don't need to use the member to use `CustomScanState`:

```c
PGrnScanState *state = palloc(sizeof(PGrnScanState));
state->customScanState.XXX = ...;
```

Reply:
I didn't know that this type of structure usage is called parent and child.
So I was misunderstanding it.

The name `parent` made sense to me.

Reply:
It's a generic `struct` usage to use object oriented programming in C.
"parent" is for "parent class".